main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -20,11 +20,11 @@ import java.lang.reflect.{Constructor, Field, Method, Modifier}
 
```

Comment:
Not really sure where this comment belongs, but I note that on page 15 in THIH (bottom) there is a long list of requirements that should be checked additionally. We should probably add these to some milestone list:

```
The Haskell report imposes some further restrictions on class and instance declarations that are not enforced by the definitions of addClass and addInst. For
example, the superclasses of a class should have the same kind as the class itself;
the parameters of any predicates in an instance context should be type variables,
each of which should appear in the head of the instance; and the type appearing in
the head of an instance should consist of a type constructor applied to a sequence
of distinct type variable arguments. Because these conditions have no direct impact
on type checking, and because they are straightforward but tedious to verify, we
have chosen not to include tests for them here, and instead assume that they have
been checked during static analysis prior to type checking.
```

I think you already check for some of these.

Reply:
* > the superclasses of a class should have the same kind as the class itself
  * Depends on superclasses and non-`*` classes
* > the parameters of any predicates in an instance context should be type variables, each of which should appear in the head of the instance
  * `TODO`
* > and the type appearing in the head of an instance should consist of a type constructor applied to a sequence of distinct type variable arguments
  * This is the complex/simple instance question from before. On my list :)